core/templates/dev/head/directives.js

```diff
@@ -156,3 +156,30 @@ oppia.directive('mobileFriendlyTooltip', ['$timeout', function($timeout) {
     }
   };
 }]);
+
+oppia.directive('closeOnClickingOutside', ['$document', function($document) {
+  return {
+    restrict: 'A',
+    link: function(scope, element, attrs) {
+      element.click(function(evt){
+        mdCard = element[0].childNodes[1];
+        if(evt.target.children[0] == mdCard) {
+          angular.element($document[0].getElementsByClassName(
+            'background-mask')).css({display:'none'});
+        }
+      });
+    }
+  }
+}]);
+
+oppia.directive('onClickViewMdCard',['$document', function($document) {
```

Comment:
Ditto, this isn't the way to go about it. How about putting the background mask container div in collection_player.html instead and adding an ng-click attr to anchor tag that toggles a variable that turns it on (using ng-class)? Again, try to find a method which doesn't use restrict: 'A'.